### PR TITLE
Increased test coverage of important classes

### PR DESCRIPTION
**Pull request by niels89**
Extended existing test suite by increasing test code coverage of important and partially untested classes.

### CODE REVIEW

**Review by Gamebuster19901 — COMMENTED** [review]
Style looks good.

**Comment by jluech** [comment]
sorry for the late adjustments, we went over the code again and found some small issues

File: tests/de/gurkenlabs/litiengine/util/geom/GeometricUtilitiesTests.java [hunk 1]
```diff
@@ -191,6 +202,278 @@ public void testDeltaX(double angle, double expectedDeltaX){
...
+
+    // assert
+    assertNull(avgPosition);
+  }
+
+  /**
+   * TODO: Sometimes, floating point precision can stand in the way of calculating accurate intersections, this
+   *  should probably be investigated. @see{@link java.awt.geom.RectangularShape#intersects(Rectangle2D)}
+   *  The inaccurate cases are marked in @see{getIntersectsEllipseArguments}
+   */
```
Comment (niels89) [inline tests/de/gurkenlabs/litiengine/util/geom/GeometricUtilitiesTests.java:353]: In this test case, we found an issue. Seems to be the same as in https://github.com/gurkenlabs/litiengine/pull/397#discussion_r625813854

**Comment by niels89** [comment]
Resolved the conflict in EntityTests

**Review by nightm4re94 — APPROVED** [review]
Thank you for these useful additions :)
We definitely need to address the intersection precision issue.

**Pull request merged**